src/site/content/en/blog/virtualkeyboard/index.md

```diff
@@ -0,0 +1,186 @@
+attribute). The method returns `undefined` but triggers a `geometrychanged` event.
+
+```js
+navigator.virtualKeyboard.show();
+```
+
+To hide the virtual keyboard, call the `hide()` method. The method returns `undefined` but triggers
+a `geometrychanged` event.
+
+```js
+navigator.virtualKeyboard.hide();
+```
+
+### Being informed of geometry changes
+
+Whenever the virtual keyboard appears or disappears, the `geometrychanged` event is dispatched. The
+event's `target` property contains the new geometry of the virtual keyboard inset as a
+[`DOMRect`](https://www.w3.org/TR/geometry-1/#domrect).
+
+```js
+navigator.virtualKeyboard.addEventListener('geometrychanged', (event) => {
+  const { x, y, width, height } = event.target;
+  console.log('Virtual keyboard geometry changed:', x, y, width, height);
+});
+```
+
+### Getting the current geometry
+
+You can get the current geometry of the virtual keyboard by looking at the `boundingRect` property.
+It exposes the current dimensions of the virtual keyboard as a
+[`DOMRect`](https://www.w3.org/TR/geometry-1/#domrect) object.
+
+```js
+const { x, y, width, height } = navigator.virtualKeyboard.boundingRect;
+console.log('Virtual keyboard geometry:', x, y, width, height);
+```
+
+### The CSS environment variables
+
+The VirtualKeyboard API exposes a set of CSS environment variables that provide information about
+the virtual keyboard's appearance.
+
+- `keyboard-inset-top`
+- `keyboard-inset-right`
+- `keyboard-inset-bottom`
+- `keyboard-inset-left`
+- `keyboard-inset-width`
+- `keyboard-inset-height`
+
+The virtual keyboard insets are six environment variables that define a rectangle by its top, right,
+bottom, and left insets from the edge of the viewport. The width and height insets are calculated
+from the remaining insets for developer ergonomics. The default value of each keyboard inset is
+`0px` if a fallback value is not provided, else it gets updated when the `boundingRect` value
+changes.
+
+### The virtual keyboard policy
+
+Not always when an editable element is focused, the virtual keyboard should appear. An example is a
+spreadsheet application where the user can tap a cell for its value to be included in a formula of
+another cell. The `virtualkeyboardpolicy` is an attribute whose keywords are the strings `auto` and
+`manual`. When specified on an element that is a `contenteditable` host, `auto` causes the
+corresponding editable element to automatically show the virtual keyboard when it is focused or
+tapped, and `manual` decouples focus and tap on the editable element from changes in the virtual
+keyboard's current state.
+
+```html
+<!-- Do nothing on regular focus, but show the virtual keyboard on double-click. -->
+<div
+  contenteditable
+  virtualkeyboardpolicy="manual"
+  inputmode="text"
+  ondblclick="navigator.virtualKeyboard.show();"
+>
+  Double-click to edit.
+</div>
+```
+
+## Demo
+
+You can see some aspects of the VirtualKeyboard API in action in a
```

Comment:
Some aspects? Do demos typically show everything? If the answer is 'no', then it's good that you asked yourself if that should be called out. I need to understand your thinking to know if this is something I should call out in other drafts.

Reply:
Our demos rarely cover _all_ aspects, so I don't think I need to call out that it only covers _some_ aspects.